
print(unknown_variable)
# NameError: name 'unknown_variable' is not defined

# print("2" + 2)
# TypeError: can only concatenate str (not "int") to str

# int("foo")
# ValueError: invalid literal for int() with base 10: 'foo'

# my_list = [1, 2, 3]
# print(my_list[3])
# IndexError: list index out of range

# my_dict = {"a": 1}
# print(my_dict["b"])
# KeyError: 'b'

# x = 5
# x.append(3)
# AttributeError: 'int' object has no attribute 'append'

# x = 1 / 0
# ZeroDivisionError: division by zero

# raise Exception("Something went wrong but I don't know why :(")
# A customized "error" occurred when the programmer manually throw error with raise keyword

